### PR TITLE
mgr/diskprediction: use the posix variant of ETIME

### PR DESCRIPTION
**Pull request by wjwithagen**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/pybind/mgr/diskprediction_cloud/common/__init__.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ class TimeoutError(Exception):
     pass
 
 
-def timeout(seconds=10, error_message=os.strerror(errno.ETIME)):
+def timeout(seconds=10, error_message=os.strerror(errno.ETIMEDOUT)):
```
Comment (tchaikov) [inline src/pybind/mgr/diskprediction_cloud/common/__init__.py]: i think ETIME is also defined by POSIX. see https://pubs.opengroup.org/onlinepubs/9699919799/functions/ioctl.html . FWIW, IEEE Std 1003.1-2017 is also known as POSIX.1-2017. i think it's just FreeBSD does not care to support XSI STREAMS (properly).

please fix the commit message.

Comment (wjwithagen) [reply src/pybind/mgr/diskprediction_cloud/common/__init__.py]: @tchaikov 
You are correct.  Most of the FreeBSD code creates a compatibility shim like:
```
/usr/local/include/libsync.h:39:#define ETIME ETIMEDOUT
```
Will amend

Comment (epuertat) [inline src/pybind/mgr/diskprediction_cloud/common/__init__.py]: I don't get the sense of using OS `errno` codes for getting an i18n'ed app error string for an exception that doesn't even inherit from `OSError`, whose value is never checked, and while everything else is not i18n'ed either.

I'd rather remove the whole `error_message=os.strerror(errno.ETIMEDOUT)` line and put the default message in the exception:
```py
class TimeoutError(Exception):
  def __init__(self, msg="Something kinda timed out somewhere"):
    super(TimeoutError, self).__init__(msg)
```

Comment (wjwithagen) [reply src/pybind/mgr/diskprediction_cloud/common/__init__.py]: And how different is this from a less invasive version?
```py
def timeout(seconds=10, error_message="Something kinda timed out somewhere"):
```

Comment (tchaikov) [reply src/pybind/mgr/diskprediction_cloud/common/__init__.py]: @wjwithagen the difference is all about readability. mainly, explicit is better than implicit.

- better readability: this `timeout()` function is never called with the `error_message` parameter.
- and better readability: msg is not connected to a certain errno.
- also better readability: TimeoutError is always constructed with this `error_message` so, let's just make this explicit.

Comment (tchaikov) [reply src/pybind/mgr/diskprediction_cloud/common/__init__.py]: i created a more invasive version at #31176

**Pull request closed without merging**